src/expression.d

```diff
@@ -7786,6 +7789,14 @@ public:
             msg = msg.implicitCastTo(sc, Type.tchar.constOf().arrayOf());
             msg = msg.optimize(WANTvalue);
         }
+        else  // no message - use assert expression as msg
+        {
+            OutBuffer buf;
+            buf.printf("%s failed", assertExpMsg);
```

Comment:
Just saw the test, I understand the choice of assert now. IMO we should just provide the condition expression, rather than the full `AssertExpression`. Does that make sense ?